services/comparison.py

```diff
@@ -701,7 +704,9 @@ def base_report(self):
     @cached_property
     def head_report(self):
         try:
-            report = report_service.build_report_from_commit(self.head_commit)
+            report = report_service.build_report_from_commit(
+                self.head_commit, report_class=ReadOnlyReport
```

Comment:
Could this be unsafe?

Reply:
I don't think the API does any report modification so this seems like a good move to me.